### ISSUE TITLE
cmon don't get killed on startup when container is started after docker daemon gets (re)started

### ISSUE THREAD

**Issue by waja** [severalnines/docker#35]
Hi there,

I'm running the following `docker-compose` setup:

```
services:
  clustercontrol:
    environment:
      CMON_PASSWORD: iu3wWqr0ewr18a
      DOCKER_HOST_ADDRESS: cc.test.org
      MYSQL_ROOT_PASSWORD: VBpd5cdpzW24Isf
    hostname: cc.test.org
    image: severalnines/clustercontrol
    networks:
      default: {}
    restart: on-failure
    volumes:
    - /srv/docker/clustercontrol/data/cmon.d:/etc/cmon.d:rw
    - /etc/localtime:/etc/localtime:ro
    - /etc/timezone:/etc/timezone:ro
    - /srv/docker/clustercontrol/data/sshkey:/root/.ssh:rw
    - /srv/docker/clustercontrol/data/backups:/root/backups:rw
    - /srv/docker/clustercontrol/data/cmonlib:/var/lib/cmon:rw
    - /srv/docker/clustercontrol/data/mysql:/var/lib/mysql:rw
version: '3'
```
This works very well.

Except one scenario: When the docker daemon gets restarted (via systemctl or at reboot) the container starts as usual, but `/entrypoint.sh` will try to kill `ctop` infinite:

```
clustercontrol_1  | >> Starting ClusterControl container [Sun Dec 19 13:59:30 CET 2021]
clustercontrol_1  | >> Controller version: 1.9.1.5006
clustercontrol_1  | >> Datadir is not empty..
clustercontrol_1  | >> Checking for lost+found existance
clustercontrol_1  | >> Ensure MySQL datadir has correct permission/ownership..
clustercontrol_1  | 
clustercontrol_1  | >> Starting MySQL daemon..
clustercontrol_1  | 
clustercontrol_1  | >> MySQL failed to start. Aborting..
clustercontrol_1  | >> Starting ClusterControl container [Sun Dec 19 13:59:35 CET 2021]
clustercontrol_1  | >> Controller version: 1.9.1.5006
clustercontrol_1  | >> Datadir is not empty..
clustercontrol_1  | >> Checking for lost+found existance
clustercontrol_1  | >> Ensure MySQL datadir has correct permission/ownership..
clustercontrol_1  | 
clustercontrol_1  | >> Starting MySQL daemon..
clustercontrol_1  | 211219 13:59:36 mysqld_safe Logging to '/var/lib/mysql/error.log'.
clustercontrol_1  | 211219 13:59:36 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
clustercontrol_1  | 
clustercontrol_1  | >> MySQL started. Looking for existing cmon/dcps data..
clustercontrol_1  | 
clustercontrol_1  | >> Found existing cmon/dcps database
clustercontrol_1  | >> Setting INITIALIZED=1
clustercontrol_1  | 
clustercontrol_1  | >> Existing token: 4135a4ac3884f89182676dac1bf8d88d79e522b1
clustercontrol_1  | 
clustercontrol_1  | >> Updating API token..
clustercontrol_1  | 
clustercontrol_1  | >> Retrieving existing cmon credentials..
clustercontrol_1  | 
clustercontrol_1  | >> Updating database credentials..
clustercontrol_1  | 
clustercontrol_1  | >> Setting up public key directory..
clustercontrol_1  | 
clustercontrol_1  | >> Updating existing dcps schema
clustercontrol_1  | 
clustercontrol_1  | >> Bootstrapping completed.
clustercontrol_1  | 
clustercontrol_1  | >> Preparing SSH daemon
clustercontrol_1  | Generating public/private rsa key pair.
clustercontrol_1  | Your identification has been saved in /etc/ssh/ssh_host_rsa_key.
clustercontrol_1  | Your public key has been saved in /etc/ssh/ssh_host_rsa_key.pub.
clustercontrol_1  | The key fingerprint is:
clustercontrol_1  | SHA256:7P/kl25xgEefl7uFY42s5ND9g45HKetvFePqGiK2HTs root@cc.test.org
clustercontrol_1  | The key's randomart image is:
clustercontrol_1  | +---[RSA 2048]----+
clustercontrol_1  | |                 |
clustercontrol_1  | |              .  |
clustercontrol_1  | |             o .o|
clustercontrol_1  | |       .    . o=o|
clustercontrol_1  | |        S  . +o==|
clustercontrol_1  | |       .  ..ooO++|
clustercontrol_1  | |       o.o ==o+*o|
clustercontrol_1  | |      . +E++==+o.|
clustercontrol_1  | |       . o+=XB. .|
clustercontrol_1  | +----[SHA256]-----+
clustercontrol_1  | Generating public/private dsa key pair.
clustercontrol_1  | Your identification has been saved in /etc/ssh/ssh_host_dsa_key.
clustercontrol_1  | Your public key has been saved in /etc/ssh/ssh_host_dsa_key.pub.
clustercontrol_1  | The key fingerprint is:
clustercontrol_1  | SHA256:9ysFPDJIXdJeSatM6OeoEKI5szt0MAn4XDzNdB0ZukQ root@cc.test.org
clustercontrol_1  | The key's randomart image is:
clustercontrol_1  | +---[DSA 1024]----+
clustercontrol_1  | |.  . +..E++=..   |
clustercontrol_1  | |o   + +o.++ o.   |
clustercontrol_1  | |.o.. o .+o...    |
clustercontrol_1  | | +o   .oo+=.     |
clustercontrol_1  | |  + .   So=o     |
clustercontrol_1  | | + o .   = ..    |
clustercontrol_1  | |* . .   . ...    |
clustercontrol_1  | |.+   . .  .  .   |
clustercontrol_1  | |oo    .    ..    |
clustercontrol_1  | +----[SHA256]-----+
clustercontrol_1  | 
clustercontrol_1  | >> Starting CMON to grant s9s CLI users..
clustercontrol_1  | The Cmon Controller is initializing.
clustercontrol_1  | Going to daemonize.. - cmon will write a log in /var/log/cmon.log from now
clustercontrol_1  | >> Generating key for s9s CLI
clustercontrol_1  | Grant user 'dba' succeeded.
clustercontrol_1  | >> Configuring s9s.conf
clustercontrol_1  | Grant user 'ccrpc' succeeded.
clustercontrol_1  | 
clustercontrol_1  | >> Testing ccrpc user..
clustercontrol_1  | >> Looks good.
clustercontrol_1  | 
clustercontrol_1  | >> Checking PID of cmon process..
clustercontrol_1  | >> PID of cmon(s): -- 588 572 --
clustercontrol_1  | 588 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 572
...
```
Entering the running container and sending a `kill -9` to the process fixes the issue, but this just no meaningfull solution.

Normal startup when starting the container (not by `dockerd`):
```
...
clustercontrol_1  | >> Testing ccrpc user..
clustercontrol_1  | >> Looks good.
clustercontrol_1  | 
clustercontrol_1  | >> Checking PID of cmon process..
clustercontrol_1  | >> PID of cmon(s): -- 595 594 593 572 --
clustercontrol_1  | 595 594 593 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 614 613 612 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 614 613 612 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 665 664 663 662 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 665 664 663 662 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 689 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 706 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 714 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 722 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 739 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | 747 572
clustercontrol_1  | >> Stopping CMON..
clustercontrol_1  | >> CMON stopped
...
```

Is there a way to fix that or I`m doing something wrong?

Thanks, Jan.

**Comment by waja** [comment]
Hi,

is this an expected behavior?

**Comment by alenloncaric** [comment]
Hi,

i have exact same behaviour, Stoping cmon until everything starts. 

But then i cannot create user:
```
We are unable to create a new Admin user.
There might be an issue with a s9s command line user.
Please open a terminal and try to run this script located at
/var/www/html/clustercontrol/app/tools/reset-ccrpc.sh
to reset the command line user needed for the admin user creation to work.

See also further troubleshooting guides here, https://docs.severalnines.com/docs/clustercontrol/troubleshooting/common-issues/#clustercontrol-controller-cmon
```

I cannot find anything... This is a fresh container

**Comment by ashraf-s9s** [comment]
Hi,

This bug is fixed in the latest image tag 1.9.7. You may also use the `CMON_STOP_TIMEOUT `variable (default is 30s) to control this behavior.
